addons/web_editor/static/lib/odoo-editor/test/spec/list.test.js

```diff
@@ -94,6 +94,13 @@ describe('List', () => {
                             `),
                         });
                     });
+                    it('should create a new unordered list if closestBlock is deeply nested inside a nav-item list', async () => {
```

Comment:
The fact that it's "deeply nested" shouldn't play a role here. It should do the same for `<ul><li class="nav-item">a[]b</li></ul>` as for `<ul><li class="nav-item"><div><p>a[]b</p></div></li></ul>`, and both should be tested.